src/helpers/types.ts

```diff
@@ -143,31 +144,94 @@ export interface AssetMetadata extends Struct {
   existentialDeposit: u128
 }
 
-export interface LoanInfo extends Struct {
+export interface LoanInfoCreated extends Struct {
   schedule: LoanRepaymentSchedule
-  collateral: LoanAsset
-  collateralValue: u128
-  valuationMethod: LoanValuationMethod
-  restrictions: LoanRestrictions
-  interestRate: u128
+  collateral: ITuple<[u64, u128]>
+  pricing: LoanPricing
+  restrictions: {
+    borrows: Enum
+    repayments: Enum
+  }
 }
 
-export interface LoanActiveInfo extends Struct {
-  loanId: u64
-  info: LoanInfo
+export interface LoanInfoActive extends Struct {
+  schedule: LoanRepaymentSchedule
+  collateral: ITuple<[u64, u128]>
+  restrictions: {
+    borrows: Enum
+    repayments: Enum
+  }
   borrower: AccountId32
-  writeOffStatus: LoanWriteOffStatus
+  writeOffPercentage: u128
   originationDate: u64
-  normalizedDebt: u128
+  pricing: LoanActivePricing
   totalBorrowed: u128
   totalRepaid: u128
+  totalRepaidUnchecked: u128
 }
 
-export interface LoanClosedInfo extends Struct {
-  closedAt: u32
-  info: LoanInfo
-  totalBorrowed: u128
-  totalRepaid: u128
+export interface LoanActivePricing extends Enum {
+  isInternal: boolean
+  isExternal: boolean
+  asInternal: LoanInternalActivePricing
+  asExternal: LoanExternalActivePricing
+}
+
+export interface LoanInternalActivePricing extends Struct {
+  info: {
+    collateralValue: u128
+    valuationMethod: LoanValuationMethod
+    interestRate: u128
+    maxBorrowAmount: LoanInternalPricingMaxBorrowAmount
+  }
+  normalizedDebt: u128
+  writeOffPenalty: u128
+}
+
+export interface LoanExternalActivePricing extends Struct {
+  info: {
+    priceId: CfgOracleKey
+    maxBorrowAmount: LoanExternalPricingMaxBorrowAmount
+  }
+  outstandingQuantity: u128
+}
+
+export interface CfgOracleKey extends Enum {
+  isIsin: boolean
+  asIsin: unknown // [u8;12]
```

Comment:
@filo87 I'm also not sure how to type this properly, can you help?

Reply:
That is a `U8aFixed` type 😉

Reply:
```suggestion
  asIsin: U8aFixed
```